src/main/java/com/paneedah/weaponlib/EntityProjectile.java

```diff
@@ -84,15 +87,15 @@ public EntityProjectile(World world, EntityLivingBase thrower, float velocity, f
     }
 
     public void setPositionAndDirection() {
+        BloodPacketClient.rotationYaw = this.rotationYaw;
 
         this.setLocationAndAngles(thrower.posX, thrower.posY + (double) thrower.getEyeHeight(),
                 thrower.posZ, compatibility.getCompatibleAimingRotationYaw(thrower), thrower.rotationPitch);
 
 
-
-        this.posX -= (double) (CompatibleMathHelper.cos(this.rotationYaw / 180.0F * (float) Math.PI) * 0.16F);
+        this.posX -= (double) (CompatibleMathHelper.cos(this.rotationYaw / 180.0F * (float) Math.PI) * (isAim ? 0f : 0.16F));
         this.posY -= 0.10000000149011612D;
-        this.posZ -= (double) (CompatibleMathHelper.sin(this.rotationYaw / 180.0F * (float) Math.PI) * 0.16F);
+        this.posZ -= (double) (CompatibleMathHelper.sin(this.rotationYaw / 180.0F * (float) Math.PI) * (isAim ? 0f : 0.16F));
```

Comment:
I think there is a little logic error here, with the ternary operator the first option is the true one, so this should be (isAim ? 0.16F : 0)

Reply:
You sure? Pretty sure you said if he's aiming there should be 0 offset, while shooting whithout aim there is an offset.

Reply:
> You sure? Pretty sure you said if he's aiming there should be 0 offset, while shooting whithout aim there is an offset.

We already resolved this in dm's

Reply:
I have always had a question, why not turn off the world and screen shaders by default? Isn't that better? Perhaps most players' computer graphics cards are not good, but they do not know that turning off these two features will greatly improve their frame rate